File: code/matlab/lab7.m
clc;
clear all;
close all;

% Solution (a)
Tn = 150;
N = 70;
delT = Tn/N;
fprintf('a) delT = %.2f micro_s\n\n',delT);

% Solution (b)
Tn = 4;
MBW = 2/delT;
fprintf('b)\nThe maximum bandwidth that the SMRCIM model can accurately represent = %.3f MHz\n',MBW);
Tn = 4;
delT = (Tn/N)*1000;
RFBW = (2/delT)*1000;
fprintf('delT for SMRCIM urban mircrocell model is %.3f ns\n',delT);
fprintf('The maximum RF bandwidth that can be represented is %.2f MHz\n\n',RFBW);

% Solution (c)
Exdel = 500;
delT = Exdel/N;
disp('c)')
fprintf('For indoor channel %f ns\n',delT);
RFBW = (2/delT)*1000;
fprintf('The maximum RF bandwidth for the indoor channel model is %.2f MHz\n',RFBW);

File: code/python/lab7.py
Max_excess_delay_Tn = 150
N = 70
Delta_T = round((Max_excess_delay_Tn / N),2)
print('(a):\nDelta Tao =',Delta_T,'us.')
print('\n(b):')
MBW = round((2/Delta_T), 3)
print('The maximum bandwidth that the SMRCIM model can accurately represent =', MBW,'MHz')
Tn = 4
New_Delta_T = Tn/N
RFBW = 2/New_Delta_T
print('The maximum RF bandwidth  = ', RFBW,'MHz')
print('\n(c):')
Excess_delay = 500
Indoor_Channel_DT = round((Excess_delay / N),2)
print('Indoor channel Delta Tao =', Indoor_Channel_DT,'ns')
RFBW_for_c = 1000*round((2 / Indoor_Channel_DT), 2)
print('The maximum RF bandwidth for the indoor channel model =', RFBW_for_c,'MHz')

